committeeoversightapp/views.py

```diff
@@ -111,18 +121,44 @@ def prepare_results(self, qs):
         edit_string = "<a href=\"{}\"><i class=\"fas fa fa-pencil-alt\" id=\"edit-icon\"></i></a>"
         delete_string = "<a href=\"{}\"><i class=\"fas fa fa-times-circle\" id=\"delete-icon\"></i></a>"
 
-        if self.request.user.is_authenticated:
-            for item in qs:
-                json_data.append([
-                    item.updated_at.strftime("%Y-%m-%d %I:%M%p %Z"),
-                    detail_string.format(
-                        escape(reverse_lazy(
-                            'detail-event',
-                            kwargs={'pk':item.pk}
-                        )),
-                        escape(item.name.title())
-                    ),
-                    item.start_date,
+        for item in qs:
```

Comment:
Hm, the logic in this loop is getting pretty tough to follow. Can you think of a way to break this into methods or roll some of this conditional string formatting back up into model properties?

Reply:
Refactored! Thanks for the nudge.